File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        vector<int>tab(n+1);

        tab[0] = nums[0];
        tab[1] = max(nums[0],nums[1]);

        for(int i=2;i<=n;i++){
            tab[i] = max(tab[i-2]+nums[i],tab[i-1]);
        }
        return tab[n-1];
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        # if there are dups go thru list if seen before then ur cooked
        # can sort then see if there are dups 
        
        mySet = set()
        for n in nums:
            if n not in mySet:
                mySet.add(n);
            else:
                return True;
        return False;


        